pkg/network/ebpf/c/sockfd.h

```diff
@@ -35,7 +35,7 @@ struct bpf_map_def SEC("maps/pid_fd_by_sock") pid_fd_by_sock = {
     .namespace = "",
 };
 
-static __always_inline void clear_sockfd_maps(struct sock* sock) {
+__attribute__((unused)) static __always_inline void clear_sockfd_maps(struct sock* sock) {
```

Comment:
❓ I see this used in `kprobe/tcp_close`

Reply:
This mitigates an issue that can occur on older kernels where `clang` can generate `Wunused-function` warnings for static inline functions defined in header files (even if they end up being used in a source file later on) -  see [this failed job](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/125110047#L738) as an example.

I'll change the attribute to `maybe_unused` & leave some comments where this is done to make it more clear

Reply:
I think those functions are actually unused in that particular `.c` file. For example: `flip_tuple` is only used in `http.c`.

We can use `maybe_unused` for now. We should have a discussion about if we want to restructure the header files to avoid this.